ProcessMaker/Jobs/DownloadSecurityLog.php

```diff
@@ -37,18 +50,161 @@ public function __construct(User $user, string $format, int $userId = null)
      * Execute the job.
      *
      * @return void
+     * @throw Exception
      */
     public function handle()
     {
-        //1. Get all data from security_logs table
-        //2. Create a file in specified format: csv or xml
-        //3. Zip this file
-        //4. Store in S3, with private visibility, with 24h of lifecycle
-        //5. Send email to user with the link or error message
-        if (mt_rand(1, 10) <= 8) {
-            event(new SecurityLogDownloadJobCompleted($this->user, true, __('Click on the link and download the file. This link will be available until midnight tonight.'), 'http://processmaker.com/?download=true'));
-        } else {
-            event (new SecurityLogDownloadJobCompleted($this->user, false, __('Sorry, it was not possible to generate the log file. Please contact the administrator.')));
+        if (!Media::s3IsReady()) {
+            event(new SecurityLogDownloadJobCompleted($this->user, false, __('Sorry, this feature requires the configured AWS S3 service.. Please contact the administrator.')));
+            return;
+        }
+        try {
+            $collection = $this->getCollection();
```

Comment:
In wrosts case scenario, how log this collection could be?
Consider than an array uses a lot of memory.
You could consider to use cursor() instead